clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -320,7 +320,7 @@ public Long validate(Object item) {
             if (item instanceof Long)
                 return (Long) item;
             else
-                throw new SchemaException(item + " is not a Long.");
+                throw new SchemaException(item + " is not an unsigned integer.");
```

Comment:
This would be inconsistent with the other messages, which refer to Java types. It's a condition on the Java type, so including it in the message is reasonable imho.
```suggestion
                throw new SchemaException(item + " is not a Long (encoding an unsigned integer).");
```

Reply:
Though now I look at the message for `UINT16` I see it would be consistent with that. Still I think because there are two types involved here, the Java type and the network type, including both is clearest.

Reply:
make sense. will copy that.